### ISSUE TITLE
[rust] relay-compiler to support template literals across lines.

### ISSUE THREAD

**Issue by maraisr** [facebook/relay#3473]
with the new rust compiler if a template literal spans more than one line it will refuse to output artifacts—this is prevliant when used with Prettier that decides its own formatting:

```ts
// ❌ No produced documents
const Thing = ({ search: searchRef }) => {
	const search = useFragment(
		graphql`
			fragment search_data on SearchItem {
				name
			}
		`,
		searchRef,
	);

	return <h1>{ search.name } < /h1>;;
};

// ✅ Produces documents
const Thing = ({ search: searchRef }) => {
	const search = useFragment(
		graphql`fragment search_data on SearchItem {
				name
			}`,
		searchRef,
	);

	return <h1>{ search.name } < /h1>;;
};
```

Like i suggested, this is more than okay to enforce — but with a lack of control to when prettier does stuff it makse it hard to enforce that?

---

More than happy to help rectify this, but will need to just be pointed at the correct files to go to impl this.

**Comment by josephsavona** [comment]
Hmm, that is not the case internally so I don't believe this is a general issue with the new compiler. Are you using Flow or TypeScript (or JS)?

**Comment by maraisr** [comment]
I'm using typescript yeah — I'll do some further digging locally and try find a root cause!

**Comment by kassens** [comment]
To avoid full JS parsing including different variants like TypeScript, Flow or latest babel features, we do a simple "tokenization". Internally we've seen 2 or 3 failures across many many files. Cases like "graphql`" appearing in another string might trip up the analyzer.

Search for "`" in your code and remove that part to find the offending part in the file. Might be useful to get a repro, but it's hard :( So far we've gotten away with this pretty successfully. Maybe your or TypeScript have different patterns that we could potentially encode.

Code for that is here: https://github.com/facebook/relay/blob/f73417979344c9e5b61dbabcc80c0414e5d495e9/compiler/crates/extract-graphql/src/lib.rs#L71

**Comment by maraisr** [comment]
Yeah can surely understand "my code broken fix it" not right! Thanks for the tips, will keep at it! 👌🏻 can probably close this issue, and will reopen if i find something! Thanks people 🥰

**Comment by maraisr** [comment]
Going to reopen this — am in a brand new space, and issue still there: Have a check at => 
https://github.com/maraisr/meros/tree/main/examples/relay-with-helix

and run `pnpm format` in the root of the monorepo.
